cmd/chart-repo/utils.go

```diff
@@ -271,13 +271,43 @@ func parseRepoIndex(body []byte) (*helmrepo.IndexFile, error) {
 	return &index, nil
 }
 
-func chartsFromIndex(index *helmrepo.IndexFile, r repo) []chart {
+func chartsFromIndex(index *helmrepo.IndexFile, r repo, filter *filters) []chart {
 	var charts []chart
+ENTRYLOOP:
 	for _, entry := range index.Entries {
 		if entry[0].GetDeprecated() {
 			log.WithFields(log.Fields{"name": entry[0].GetName()}).Info("skipping deprecated chart")
 			continue
 		}
+		if len(filter.Annotations) > 0 {
+			for a, av := range filter.Annotations {
+				if v, ok := entry[0].Annotations[a]; ok {
+					if len(av) == 0 {
+						charts = append(charts, newChart(entry, r))
+						continue ENTRYLOOP
```

Comment:
I think we can avoid this and improve the readability here by splitting out these filters into separate functions. Also, there might be a bug here. What if I have a name filter and an annotation filter, only the annotation filter will hold?

Something like:

```
if !matchesAnnotationsFilter(filter.Annotations, entry) || !matchesNamesFilter(filter.Names, entry) {
  // log skipping chart
  continue
}

charts = append(charts, newChart(entry, r))
```

Those functions can return true if filter.Annotations or filter.Names is empty.

WDYT?

Reply:
Done, I was thinking the same thing when using the ENTRYLOOP label but was being lazy.  Since my intention was to sync if any of the filters matched I did a single function that returns true if a filter matched.